jax/numpy/lax_numpy.py

```diff
@@ -2151,7 +2163,7 @@ def array(object, dtype=None, copy=True, order="K", ndmin=0):
     raise TypeError("Unexpected input type for array: {}".format(type(object)))
 
   if ndmin > ndim(out):
-    out = lax.reshape(out, (1,) * (ndmin - ndim(out)) + shape(out))
+    out = broadcast_to(out, (1,) * (ndmin - ndim(out)) + shape(out))
```

Comment:
For broadcasting in leading dimensions you could simply write:
`lax.broadcast(out, (1,) * (ndmin - ndim(out))`